src/center.rs

```diff
@@ -61,12 +61,14 @@ pub struct Center {
 //--- Actions
 
 /// Add a zone.
-pub fn add_zone(
+pub async fn add_zone(
     center: &Arc<Center>,
     name: Name<Bytes>,
     policy: Box<str>,
     source: api::ZoneSource,
 ) -> Result<(), ZoneAddError> {
+    register_zone(center, name.clone(), policy.clone()).await?;
```

Comment:
Hmm, yeah I guess this makes sense. I somehow didn't realize we could just make the keys before making the zone. Nice!

Reply:
I honestly don't know what the best order is, there just things that need to happen before a zone can be used, but I think we should also roll back partial changes if they don't all succeed and we lack that at present. There's also an overlap with the `zone enable` feature added by @bal-e in the `demo` branch yesterday that needs some consideration how to merge the two things.